### DISCUSSION TITLE
Semantic versioning && Semantic release on pivpn

### DISCUSSION THREAD

**Discussion by coolapso** [pivpn/pivpn#1402] [Q&A]
Currently updating the "pivpn management" scripts is no easy feat, specially due to breaking changes we've introduced and might introduce in the feature. 

Should pivpn start being release with semantic versioning and following [semantic-release](https://github.com/semantic-release) commit types in order to have this releases automated? 

Think this way we could make the Script check for the major version and either bail or inform the user properly about what actions to take. 

Any thoughts? @orazioedoardo @MichaIng

**Comment by MichaIng** [comment]
You mean that after an update, there may be configs missing, old obsolete ones left, or settings invalidated and such? So your idea is go use semantic versioning to differentiate between releases/updates which require migration steps and those which do not? Sounds reasonable 👍, though I lack insights of the kind of breaking changes you did or plan to do.

Generally, in case of config files, a setting which contains the config file's version could be implemented and then incremental migration steps either via updater script or via main script when parsing the config file.

For databases it is often done similarly with a database version key, like phpBB does it [here](https://github.com/phpbb/phpbb/tree/master/phpBB/phpbb/db/migration/data), but does not apply to PiVPN.

At DietPi we do incremental patches for every version increment to the system, configs, software reinstalls and such, when automatic migration or backwards compatibility within the end user scripts is difficult or unreasonable to maintain: https://github.com/MichaIng/DietPi/blob/dev/.update/patches
This includes also informing users about important changes or chances, available software updates and/or giving them choices to apply a patch which isn't mandatory but reasonable for the majority of cases.

**Comment by TinCanTech** [comment]
My next will be transitioned. It's a reasonably well established standard, so why not ?
